### DISCUSSION TITLE
Running hundreds of tests in Parallel

### DISCUSSION THREAD

**Discussion by JeffVM-Solutions** [seleniumbase/SeleniumBase#949] [Q&A] [answered]
Do you have a suggestion on if SeleniumBase would be a good solution to run hundreds of selenium tests in parallel? My company hosts and needs to do end-to-end testing on over 200 web pages. I'm setting up my own Selenium Grid but was wondering if SeleniumBase was designed for a larger number of testing scripts?

**Comment by mdmintz** [comment, marked as answer]
Hello @DevOps-Martins, SeleniumBase works well for that. You can multithread the tests via pytest with:
``pytest -n=8`` (for example) to spin up that many parallel processes. And you can combine that with Selenium Grid usage. There's a ReadMe for that here: https://seleniumbase.io/seleniumbase/utilities/selenium_grid/ReadMe/ 
You can use your own grid, or connect to another grid of your choice.
